using-packages.md

```diff
@@ -0,0 +1,262 @@
+
+### Dependencies on unpublished packages
+
+Packages can be used even when not published on Pub. For private plugins not
+intended for public publishing, or for packages not yet ready for publishing,
+additional dependency options are avaialble:
+
+* **Path** dependency: A Flutter app can depend on a plugin via a file system
+ `path:` dependency. The path can be either relative, or absolute. For example, to
+ depend on a plugin 'plugin1' located in a directory next to the app, use this
+ syntax:
+```dart
+dependencies:
+    plugin1:
+      path: ../plugin1/
+```
+
+* **Git** dependency: You can also depend on a package stored in a Git
+ repository. The package must be located in the root of the repo. Use this
+ syntax:
+```dart
+dependencies:
+    plugin1:
+      git:
+        url: git://github.com/flutter/plugin1.git
+```
+
+For more details, see the [Pub Dependencies article](https://www.dartlang.org/tools/pub/dependencies).
+
+## Examples
+
+### Example: Using the CSS Colors package {#css-example}
+
+The [URL Launcher](https://pub.dartlang.org/packages/url_launcher) plugin
+enables you to open the default browser on the mobile platform to display a
+given URL. It demonstrates how packages may also contain platform-specific code
+(we call these packages 'plugins'). It is supported on both Android and iOS.
+
+To use this plugin:
+
+1. Create a new project called 'cssdemo'
+
+1. Open `pubspec.yaml`, and replace:
+    ```dart
+    dependencies:
+      flutter:
+        sdk: flutter
+    ```
+    with:
+
+    ```dart
+    dependencies:
+      flutter:
+        sdk: flutter
+      css_colors: ^1.0.0
+    ```
+
+1. Run `flutter packages get` in the terminal, or click 'Packages get' in IntelliJ
+
+1. Open `lib/main.dart` and replace its full contents with:
+    ```dart
+    import 'package:flutter/material.dart';
+    import 'package:css_colors/css_colors.dart';
+
+    void main() {
+      runApp(new MyApp());
+    }
+
+    class MyApp extends StatelessWidget {
+      @override
+      Widget build(BuildContext context) {
+        return new MaterialApp(
+          home: new DemoPage(),
+        );
+      }
+    }
+
+    class DemoPage extends StatelessWidget {
+      @override
+      Widget build(BuildContext context) {
+        return new Scaffold(
+          body: new Container(color: CSSColors.orange)
+        );
+      }
+    }
+    ```
+
+1. Run the app. When you click the 'Show Flutter homepage' you should see the
+phone's default browser open, and the Flutter homepage appear.
+
+
+
+### Example: Using the URL Launcher plugin to launch the browser {#url-example}
+
+The [URL Launcher](https://pub.dartlang.org/packages/url_launcher) plugin
+enables you to open the default browser on the mobile platform to display a
+given URL. It demonstrates how packages may also contain platform-specific code
+(we call these packages 'plugins'). It is supported on both Android and iOS.
```

Comment:
I think this help delivered that very message; if you are a consumer there is no difference.